cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/set_asset_claimer.rs

```diff
@@ -0,0 +1,217 @@
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Tests related to claiming assets trapped during XCM execution.
+
+use crate::imports::*;
+use emulated_integration_tests_common::{
+	accounts::{ALICE, BOB},
+	impls::AccountId32,
+};
+
+use crate::imports::bhw_xcm_config::{LocationToAccountId, UniversalLocation};
+use frame_support::{assert_ok, sp_runtime::traits::Dispatchable};
+use xcm_executor::traits::ConvertLocation;
+
+#[test]
+fn test_set_asset_claimer_within_a_chain() {
+	let (alice_account, alice_location) = account_and_location(ALICE);
+	let (bob_account, bob_location) = account_and_location(BOB);
+
+	let amount_to_send = 16_000_000_000_000;
+	let assets: Assets = (Parent, amount_to_send).into();
+
+	let alice_balance_before =
+		<AssetHubWestend as Chain>::account_data_of(alice_account.clone()).free;
+	AssetHubWestend::fund_accounts(vec![(alice_account.clone(), amount_to_send * 2)]);
+	let alice_balance_after =
+		<AssetHubWestend as Chain>::account_data_of(alice_account.clone()).free;
+	assert_eq!(alice_balance_after - alice_balance_before, amount_to_send * 2);
+
+	let test_args = TestContext {
+		sender: alice_account.clone(),
+		receiver: bob_account.clone(),
+		args: TestArgs::new_para(
+			bob_location.clone(),
+			bob_account.clone(),
+			amount_to_send,
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = SystemParaToParaTest::new(test_args);
+	type RuntimeCall = <AssetHubWestend as Chain>::RuntimeCall;
+	let local_xcm = Xcm::<RuntimeCall>::builder_unsafe()
+		.set_asset_claimer(bob_location.clone())
+		.withdraw_asset(test.args.assets.clone())
+		.clear_origin()
+		.build();
+
+	AssetHubWestend::execute_with(|| {
+		assert_ok!(RuntimeCall::PolkadotXcm(pallet_xcm::Call::execute {
+			message: bx!(VersionedXcm::from(local_xcm)),
+			max_weight: Weight::from_parts(4_000_000_000_000, 300_000),
+		})
+		.dispatch(test.signed_origin));
+	});
+
+	let balance_after_trap =
+		<AssetHubWestend as Chain>::account_data_of(alice_account.clone()).free;
+	assert_eq!(alice_balance_after - balance_after_trap, amount_to_send);
+
+	let bob_balance_before = <AssetHubWestend as Chain>::account_data_of(bob_account.clone()).free;
+	let test_args = TestContext {
+		sender: bob_account.clone(),
+		receiver: alice_account.clone(),
+		args: TestArgs::new_para(
+			alice_location.clone(),
+			alice_account.clone(),
+			amount_to_send,
+			assets.clone(),
+			None,
+			0,
+		),
+	};
+	let test = SystemParaToParaTest::new(test_args);
```

Comment:
same here, you could remove `TestContext` and `SystemParaToParaTest`, just use assets instead of test.args.assets, etc, below

Reply:
removed